src/Models/Sitemap.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Rapidez\Sitemap\Models;
+
+use Illuminate\Support\Facades\Cache;
+use Rapidez\Core\Models\Model;
+use SimpleXMLElement;
+
+class Sitemap extends Model
+{
+    protected $table = 'sitemap';
+
+    public static function getCachedByStoreId(): ?array
+    {
+        $cacheKey = 'sitemaps.'.config('rapidez.store');
+
+        return Cache::rememberForever($cacheKey, function () {
+            return self::where('store_id', config('rapidez.store'))
+                ->get()
+                ->flatMap(fn ($sitemap) => self::getSitemapsFromIndex($sitemap->toArray()))
+                ->toArray();
+        });
+    }
+
+    protected static function getSitemapsFromIndex(array $sitemap): array
```

Comment:
Don't we need to make this function recursive? What if the magento sitemap already is an index?

Reply:
We already do, see https://github.com/rapidez/sitemap/pull/1#discussion_r1827666292.
We do this in the return of line 48.